src/main/java/uk/gov/hmcts/darts/dailylist/service/DailyListProcessorImpl.java

```diff
@@ -0,0 +1,187 @@
+    private final CourthouseRepository courthouseRepository;
+    private final CaseRepository caseRepository;
+    private final HearingRepository hearingRepository;
+
+    private final ObjectMapper objectMapper;
+
+    @Override
+    public void processAllDailyLists(LocalDate date) {
+        List<CourthouseEntity> allCourthouses = courthouseRepository.findAll();
+        for (CourthouseEntity allCourthouse : allCourthouses) {
+            List<DailyListEntity> dailyLists = dailyListRepository.findByCourthouse_IdAndStatusAndStartDateOrderByPublishedTimestampDesc(
+                    allCourthouse.getId(),
+                    String.valueOf(JobStatusType.NEW),
+                    date
+            );
+
+            // Daily lists are being ordered descending by date so first item will be the most recent version
+            if (!dailyLists.isEmpty()) {
+                try {
+                    processDailyList(dailyLists.get(0));
+                } catch (JsonProcessingException e) {
+                    throw new DartsApiException(DailyListError.FAILED_TO_PROCESS_DAILYLIST, e);
+                }
+
+            }
+            if (dailyLists.size() > 1) {
+                ignoreOldDailyList(dailyLists.subList(1, dailyLists.size()));
+            }
+        }
+    }
+
+    private void ignoreOldDailyList(List<DailyListEntity> dailyLists) {
+        for (DailyListEntity dailyList : dailyLists) {
+            dailyList.setStatus(String.valueOf(JobStatusType.IGNORED));
+        }
+    }
+
+    @SuppressWarnings("checkstyle:VariableDeclarationUsageDistance")
+    private void processDailyList(DailyListEntity dailyListEntity) throws JsonProcessingException {
+        DailyList dailyList = objectMapper.readValue(dailyListEntity.getContent(), DailyList.class);
+        JobStatusType statusType = JobStatusType.PROCESSED;
+
+        for (CourtList courtList : dailyList.getCourtLists()) {
+
+            String courtHouseName = courtList.getCourtHouse().getCourtHouseName();
+            Optional<CourthouseEntity> foundCourthouse = courthouseRepository.findByCourthouseNameIgnoreCase(
+                    courtHouseName);
+
+            if (foundCourthouse.isPresent()) {
+                List<Sitting> sittings = courtList.getSittings();
+                for (Sitting sitting : sittings) {
+                    List<Hearing> hearings = sitting.getHearings();
+                    for (Hearing dailyListHearing : hearings) {
+                        CourtCaseEntity courtCase = retrieveCoreObjectService.retrieveOrCreateCase(
+                                courtHouseName,
+                                dailyListHearing.getCaseNumber()
+                        );
+                        HearingEntity hearing = retrieveCoreObjectService.retrieveOrCreateHearing(
+                                courtHouseName,
+                                String.valueOf(sitting.getCourtRoomNumber()),
+                                dailyListHearing.getCaseNumber(),
```

Comment:
i think the URN/casenumber logic should be here

Reply:
ACCEPT